src/WCSimDetectorConfigs.cc

```diff
@@ -20,7 +20,10 @@
 
 void WCSimDetectorConstruction::SetSuperKGeometry()
 {
-  WCSimPMTObject * PMT = CreatePMTObject("PMT20inch", "glassFaceWCPMT");
+  WCDetectorName = "SuperK";
```

Comment:
Yes, I like this idea. I could do it now or wait until the next step where the digitizer is able to read out from various hit collections. Then I could put the hit collection which is recorded by the digitizer (which will also contain the detector name) into the geometry tree.